src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -914,9 +984,22 @@ public FormValidation doCheckUseInstanceProfileForCredentials(@QueryParameter bo
             return FormValidation.ok();
         }
 
-        public FormValidation doCheckPrivateKey(@QueryParameter String value) throws IOException, ServletException {
+        public FormValidation doCheckSshKeysCredentialsId(@QueryParameter String value) throws IOException, ServletException {
+
```

Comment:
Now that this method is looking up system credentials, it should have a permission check added.